ci_framework/roles/ci_network/tasks/apply.yml

```diff
@@ -0,0 +1,24 @@
+---
+- name: "Apply network {{ network.key }}"
+  become: true
+  when:
+    - network.value.config_nm | default(true) | bool
+  community.general.nmcli:
+    conn_name: "{{ network.value.connection | default(network.key) }}"
+    ifname: "{{ network.value.iface }}"
+    type: "{{ network.key is match('^vlan.*') | ternary('vlan', 'ethernet') }}"
+    ip4: "{{ network.value.ip4 }}"
+    mac: "{{ network.value.mac | default(omit) }}"
+    dns4: "{{ network.value.dns4 | default(omit) }}"
+    # Needed for some specific conditions
+    gw4: "{{ network.value.gw4 | default(omit) }}"
+    # Needed only for IPv6 support
+    ip6: "{{ network.value.ip6 | default(omit) }}"
+    # Needed only for IPv6 support
+    gw6: "{{ network.value.gw6 | default(omit) }}"
+    # Needed for VLAN support
+    parent_iface: "{{ network.value.parent_iface | default(omit)}}"
+    # Needed for some specific infra
+    mtu: "{{ network.value.mtu | default(1500) }}"
```

Comment:
Are we assuming that the MTU is not always passed? I coded the current openstack logic to pass the MTU always, I don't expect others to not continuing doing so.

Reply:
imho we shouldn't enforce MTU argument. Any reason to enforce it?

Reply:
maybe in that role is not need, but with the current implementation is always filled with the real MTU from the link

Reply:
We may reconsider if needed, but keeping it optional seems legit for now